concent_api/utils/tests/test_helpers.py

```diff
@@ -81,3 +87,23 @@ def test_join_messages_should_return_joined_string_separeted_with_whitespace(sel
 
     def test_join_messages_with_single_argument_should_return_single_string(self):
         self.assertEqual(join_messages('Error in Golem Message.'), 'Error in Golem Message.')
+
+    @override_settings(
+        DOWNLOAD_LEADIN_TIME=constants.DOWNLOAD_LEADIN_TIME
+    )
```

Comment:
These two tests are not enough to give us confidence that we will notice when the implementations go out of sync. Please check all the important cases:
- `maximum_download_time()`
    - `CUSTOM_PROTOCOL_TIMES` on or off
    - `rate` given explicitly or left out
    - `DOWNLOAD_LEADIN_TIME` and `MINIMUM_UPLOAD_RATE` matching the values from golem-messages or not.
    - A few different combinations of `size` and `rate`
- `subtask_verification_time()`
    - `CUSTOM_PROTOCOL_TIMES` on or off
    - A few different computation times in TTC
    - A few different results from `maximum_download_time()`

Also, please make these tests more compact. We're testing two extremely simple functions that just accept numbers and spit out the result of a short mathematical expression. I think it's bad if testing a new combination requires you to add ten lines of tests. With simple mathematical functions like that we should make it cheap to test a dozen sets of inputs and expectations in one go. You could use a loop or any other mechanism suggested by @dybi. Just make it easy to cover many cases - we badly need it here.

Reply:
Not if `CUSTOM_PROTOCOL_TIMES` off. I really meant to check with various values of these two settings. But the two most important cases are: when the match and when they do not match the built-in values.